cedar-policy-core/src/parser/loc.rs

```diff
@@ -16,6 +16,14 @@
 
 use std::sync::Arc;
 
+/// Represents an optional `Loc`
+#[cfg(not(feature = "fast-parsing"))]
+pub type MaybeLoc = Option<Loc>;
+
+/// Represents an optional `Loc`
+#[cfg(feature = "fast-parsing")]
+pub type MaybeLoc = Option<Box<Loc>>;
```

Comment:
Perhaps a comment here about why `Box` is actually faster.  I know it's in the PR description but would probably be valuable to immortalize here in code comments as well